cpp/oneapi/dal/algo/pca/backend/gpu/infer_kernel_dpc.cpp

```diff
@@ -14,76 +14,58 @@
 * limitations under the License.
 *******************************************************************************/
 
-#include <daal/src/algorithms/pca/transform/oneapi/pca_transform_dense_default_batch_oneapi.h>
-#include <daal/include/algorithms/pca/transform/pca_transform_types.h>
-
-#include "oneapi/dal/algo/pca/backend/common.hpp"
 #include "oneapi/dal/algo/pca/backend/gpu/infer_kernel.hpp"
-#include "oneapi/dal/backend/interop/error_converter.hpp"
-#include "oneapi/dal/backend/interop/table_conversion.hpp"
+#include "oneapi/dal/algo/pca/backend/common.hpp"
+#include "oneapi/dal/backend/primitives/lapack.hpp"
+#include "oneapi/dal/backend/primitives/reduction.hpp"
+#include "oneapi/dal/backend/primitives/stat.hpp"
+#include "oneapi/dal/backend/primitives/utils.hpp"
+#include "oneapi/dal/backend/primitives/blas.hpp"
 #include "oneapi/dal/table/row_accessor.hpp"
-#include "oneapi/dal/backend/interop/common_dpc.hpp"
 
 namespace oneapi::dal::pca::backend {
 
+namespace pr = oneapi::dal::backend::primitives;
+
 using dal::backend::context_gpu;
 using model_t = model<task::dim_reduction>;
 using input_t = infer_input<task::dim_reduction>;
 using result_t = infer_result<task::dim_reduction>;
 using descriptor_t = detail::descriptor_base<task::dim_reduction>;
 
-namespace daal_pca_tr = daal::algorithms::pca::transform;
-namespace daal_pca_tr_oneapi = daal::algorithms::pca::transform::oneapi;
-namespace interop = dal::backend::interop;
-
-template <typename Float>
-using daal_pca_transform_oneapi_kernel_t =
-    daal_pca_tr_oneapi::internal::TransformKernelOneAPI<Float, daal_pca_tr::Method::defaultDense>;
-
-template <typename Float>
-static result_t call_daal_kernel(const context_gpu& ctx,
-                                 const descriptor_t& desc,
-                                 const table& data,
-                                 const model_t& model) {
+template <typename Float, typename Task>
```

Comment:
for consistency remove Task from template and use input_t, result_t (instead of infer_result<Task>)